_posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc

```diff
@@ -1074,12 +1235,20 @@ setup a subscriber to the SNS topics that are created.
 
 [[configure_networking]]
 === Configure networking
-If you're using Gruntwork's link:https://github.com/gruntwork-io/module-vpc[VPC module] for your VPCs, three
+
+If you're using Gruntwork's link:https://github.com/gruntwork-io/terraform-aws-vpc[VPC module] for your VPCs, two
 of the four recommendations in this section are already taken care of! By default, none of our modules allow
-access to ports 22 or 3389 from the world, and our architecture has a least-privileges-based routing
+security groups to access to ports 22 or 3389 from the world, and our architecture has a least-privileges-based routing
 configuration by default.
 
-The only necessary step here is to run the link:https://github.com/gruntwork-io/cloud-nuke[`cloud-nuke
+To meet the 5.1 recommendation, you need to create your Network ACL Rules for the VPC-App using the
+link:https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/tree/master/modules/vpc-app-network-acls[`vpc-app-network-acls`
+module] and your NACL Rules for the VPC-Mgmt (which is being deprecated) using the
+link:https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/tree/master/modules/vpc-mgmt-network-acls[`vpc-mgmt-network-acls`
+module].
```

Comment:
Code examples plz!

Reply:
@brikis98 @marinalimeira This one is a bit tricky... For the code examples, we need to:

1. Possibly link to this section: https://www.gruntwork.io/guides/compliance/how-to-achieve-cis-benchmark-compliance/#wrapper_modules
2. Explain how we extend the `vpc-app` and `vpc-mgmt` modules by creating new ones in `infrastructure-modules`.
3. Explain how in the new `vpc-app` and `vpc-mgmt` modules we need to invoke the `vpc-mgmt-network-acls` module.
4. Explain how to create a `terragrunt.hcl` file that invokes our new VPC modules in `infrastructure-modules`.

Is there an easier way to do this? Could we just do the last step?

Reply:
I think the real solution is to add a VPC service to the CIS Service Catalog, which would include `vpc-app` and `vpc-app-network-acls`, and deploy that with one `terragrunt.hcl`. We have the ticket for this already: https://gruntwork.atlassian.net/browse/IAC-1643... But it's not part of the CIS v1.3 project.

So I guess it's a trade-off between:

1. Updating this guide with a bunch of steps that will take some time to put together and will be deprecated soon.
2. Doing the VPC service work now, and updating the guide to use that. But note that the VPC service would have to extend the existing VPC service in `terraform-aws-service-catalog`, and not all our CIS customers have access to that yet, so that would be yet another yak to shave.
3. Doing some minimal version of (1). That is, we say vaguely something like, "You should use the `vpc-app-network-acls` module to meet the new CIS v1.3 requirements. Check out our [VPC guide](https://gruntwork.io/guides/networking/how-to-deploy-production-grade-vpc-aws/) for how to deploy a VPC with the `vpc-app` module. Once that's deployed, add the `vpc-app-network-acls` module as follows: <snippet>. Deploy all this using Terragrunt: <snippet>."

It's not ideal, but (3) is probably the way to go, and we'll replace it fairly soon when the CIS Service Catalog work is done.

Reply:
I am not sure if I am following... What is the difference between (2) and [the cis-infra-live-acme](https://github.com/gruntwork-io/cis-infrastructure-live-acme/tree/master/dev/us-east-2/dev/vpc) + cis-infra-modules?

Reply:
The old way:

1. Customer has an `infra-modules` repo with a bunch of Terraform code in it.
2. Customers' `infra-live` repo sets the `source` URL to their own `infra-modules` repo.

The new way:

1. Gruntwork has an `service-catalog` repo with a bunch of Terraform code in it.
2. Customers' `infra-live` repo sets the `source` URL to Gruntwork's `service-catalog` repo.

The CIS Service Catalog project is moving us to the new way for CIS... But we're not there yet. So we're trying to decide if we want to do the work of showing customers the  "old way" for this VPC NACL stuff, only to throw it away a little later (option 1)... Or if we can go straight to the new way (option 2). Or some minimal version of the old way, as a placeholder until we finish the Service Catalog stuff (option 3).

Reply:
I'd vouch with respect for time and simplicity, that we go with the old way. And then as part of CIS Service Catalog work, we come back here to update this.

Reply:
I'll work on the `Updating this guide with a bunch of steps that will take some time to put together and will be deprecated soon.`!